frontend/src/containers/ConfigStep/DateRangePicker/validation.ts

```diff
@@ -0,0 +1,54 @@
+import { BasicConfigState } from '@src/context/config/configSlice';
+import dayjsSameOrBeforePlugin from 'dayjs/plugin/isSameOrBefore';
+import dayjsSameOrAfterPlugin from 'dayjs/plugin/isSameOrAfter';
+import dayjs, { Dayjs } from 'dayjs';
+
+dayjs.extend(dayjsSameOrBeforePlugin);
+dayjs.extend(dayjsSameOrAfterPlugin);
+
+// some SIGNIFICANT prerequisites :
+// dayjs(undefined).isValid() true
+// dayjs(null).isValid() false
+// dayjs('').isValid() false
+// if dayjsA or dayjsB either is invalid
+// all comparisons e.g. dayjsA.isBefore(dayjsB) always return false
+
+export const calculateLastAvailableDate = (date: Dayjs, coveredRange: BasicConfigState['basic']['dateRange']) => {
```

Comment:
BasicConfigState['basic']['dateRange'] should be a `Array<{start, end....}>`

Reply:
should defined a basic type

Reply:
ok